src/main/steps/application-within-proceedings/document-upload/postController.ts

```diff
@@ -23,29 +23,17 @@ export default class UploadDocumentController extends PostController<AnyObject>
     const { awp_application_form } = (req?.files as Record<string, any>) ?? {};
     const { applicationType, applicationReason } = req.params;
 
-    if (
-      req.body.onlyContinue &&
-      req.session.userCase.awp_uploadedApplicationForms &&
-      req.session.userCase.awp_uploadedApplicationForms.length > 0
-    ) {
+    const isSupportingDocuments = req.route.path === APPLICATION_WITHIN_PROCEEDINGS_SUPPORTING_DOCUMENT_UPLOAD;
+    const areFilesUploaded = isSupportingDocuments
+      ? req.session.userCase.awp_uploadedSupportingDocuments &&
```

Comment:
you can do this in 1 condition with optional property checking

Reply:
Using one condition for this now